subprojects/plugin-use/src/main/java/org/gradle/plugin/management/internal/autoapply/DefaultAutoAppliedPluginHandler.java

```diff
@@ -42,12 +43,13 @@ public PluginRequests mergeWithAutoAppliedPlugins(PluginRequests initialRequests
         }
         Project project = (Project) pluginTarget;
 
-        List<PluginRequestInternal> merged = Lists.newArrayList(initialRequests);
+        List<PluginRequestInternal> merged = Lists.newArrayList();
```

Comment:
Should take the expected size from the initial requests to avoid later array resizing.

Reply:
Can you explain the impact of providing the size of the initial request on performance?

```
List<PluginRequestInternal> merged = new ArrayList<PluginRequestInternal>(initialRequests.size());
// add auto-applied plugin requests
// add initial plugin requests
```

How significant is such optimization? Trying to understand it for future changes.

Reply:
You know both the size of the initial and auto-applied requests. By adding them up you can create an array that is guaranteed to fit both.

If you don't do that, you'll end up creating an array that might be too small, which means it'll have to be resized, resulting in more garbage.

Reply:
I was saying "unknown" here as we are also filtering the plugin requests that have been applied or requested. I made the change. Can you have a another look?

Back to my initial question: Does array resizing affect performance significantly?